lib/ahoy/tracker.rb

```diff
@@ -5,6 +5,7 @@ class Tracker
     UUID_NAMESPACE = "a82ae811-5011-45ab-a728-569df7499c5f"
 
     attr_reader :request, :controller
+    attr_accessor :user
```

Comment:
Instead of adding a `user=` method, I think it'd be better to allow a `user` option to be passed to the constructor.

Reply:
Actually I have a use case with an authentication service where
- The user is at first not logged in and I track a sign in attempt with extra context
- After authentication is successful, I make redirections or I perform specific business logic (such as checking whether account is restricted) and I may not allow the authentication to perform successfully despite having identified which user it is. In this case, I would have used something like `service.ahoy.user = user` while still not making the `current_user` available in the controller

However if you still feel this is too edgy a corner case, I can deal with this differently on my side, and remove the accessor here, only keeping it in the initializer as you suggested.